### DISCUSSION TITLE
Pytest-watch with coverage (If not existent)

### DISCUSSION THREAD

**Discussion by brunolnetto** [pytest-dev/pytest#12184] [Q&A]
#### What's the problem this feature will solve?

I can:
1. run tests with package `pytest-cov` with command run `coverage run --rcfile=.coveragerc -m pytest`. 
2. generate coverage with command run `coverage report --show-missing`. 
3. watch tests with command run `ptw --quiet --spool 200 --clear --nobeep --config pytest.ini --ext=.py --onfail="echo Tests failed, fix the issues" -v`

I would like to have a way to watch coverage while testing as well. May you enlight me?

#### Describe the solution you'd like

I wrote following bash script ith desired solution:

```
#!/bin/bash
clear

while true; do
  coverage run --rcfile=.coveragerc -m pytest
  coverage report --show-missing
  
  sleep 5  # Adjust delay between test runs if needed
  clear
done
```

#### Alternative Solutions

I have not found the proper way to proceed.

**Comment by The-Compiler** [comment]
Looks like `ptw` has a `--runner` option which should let you use `coverage run pytest`. Or, if you are indeed using the `pytest-cov` plugin, all you should need to do is pass the `--cov` argument to pytest (no need for `coverage run`).

Converting this to a discussion, since there's nothing actionable for pytest core here.

**Comment by brunolnetto** [comment]
Thanks for the quick answer. As far I understand, add a runner command should do the trick. However, it does not.

```
ptw --quiet --spool 200 --clear --nobeep \
	--config pytest.ini --ext=.py \
	--onfail="echo Tests failed, fix the issues" \
	--runner "coverage report --show-missing"
```

@The-Compiler  also mentioned about running command `pytest --cov`, which leads to error below. I suppose, there is some in between step I fail to perform.

```
ERROR: usage: pytest [options] [file_or_dir] [file_or_dir] [...]
pytest: error: unrecognized arguments: --cov
  inifile: /home/brunolnetto/github/API/pytest.ini
  rootdir: /home/brunolnetto/github/API
```
